### PR TITLE
Gonum redesign vgdraw

### PR DESCRIPTION
**Pull request by sbinet**
- add vgdraw.Canvas (in replacement of plot.DrawArea)
- introduce vg.Inch, vg.Centimeter and vg.Millimeter
- introduce plot.Ticker interface
- introduce plot.Normalizer interface
- support for persistency

### CODE REVIEW

**Comment by kortschak** [comment]
These commits don't seem to happen in order. From what I can see 3465f964e9aef09856a4aa6388a0a8d2e2263a44 should happen before dcf6a66a88c002c7400edb1d331d2f405fe2cc62, but it is presented later in the list. Do you know what is causing that?

**Comment by sbinet** [comment]
some re-ordering might have been caused by the multiple cherry-picking + rebase-fixing/merge-conflict-fixing I had to apply.
that said, the ordered list presented in this PR seems like the correct timeline (at least, the one I recall.)

**Comment by kortschak** [comment]
I'm following the graph. That makes more sense to me.

**Comment by kortschak** [comment]
I think draw would be a better name than vgdraw, but I like these changes over all. Nicely presented too.

**Comment by sbinet** [comment]
I wouldn't mind changing `vgdraw` into `draw` but it could shadow the stdlib package `image/draw` (and the upcoming `x/image/draw`)

**Comment by sbinet** [comment]
PTAL

**Comment by kortschak** [comment]
That should not be an issue. People can rename if they collide.

**Comment by sbinet** [comment]
ok. done.

File: axis.go [hunk 1]
```diff
@@ -196,29 +219,29 @@ func (a *horizontalAxis) draw(da DrawArea) {
...
-	for _, t := range a.Tick.Marker(a.Min, a.Max) {
+	for _, t := range a.Tick.Marker.Ticks(a.Min, a.Max) {
 		if t.IsMinor() {
 			continue
 		}
 		w := a.Tick.Label.Width(t.Label)
 		box := GlyphBox{
 			X:    a.Norm(t.Value),
-			Rect: Rect{Point{X: -w / 2}, Point{X: w}},
+			Rect: draw.Rect{vgdraw.Point{X: -w / 2}, vgdraw.Point{X: w}},
```
Comment (kortschak) [inline axis.go]: Missed change. Others at axis.go:327, plot.go:182,207,263.

Comment (sbinet) [inline axis.go]: Arf. Right. sed replaces only one occurrence per line.
Will fix and squash.

**Comment by sbinet** [comment]
PTAL

File: vg/len.go [hunk 2]
```diff
@@ -26,42 +13,19 @@ func Points(pt float64) Length {
...
-}
-
-// Millimeters returns a length for the given number of millimeters.
-func Millimeters(mm float64) Length {
-	return Length(mm * ptMillimeter)
-}
+// Common lengths.
+const (
+	Inch       Length = 72
+	Centimeter        = Inch / 2.54
```
Comment (kortschak) [inline vg/len.go]: Missed this, but it's spelled in American in gonum/unit, so we should leave it.

**Comment by kortschak** [comment]
LGTM, but waiting for others.

**Comment by sbinet** [comment]
@eaburns I suspect (hope!) this PR already has your LGTM, right? :)

**Comment by sbinet** [comment]
also: I should probably squash before this gets merged.

**Comment by eaburns** [comment]
LGTM, assuming that it's the same as before. I didn't look at it again (sorry, lots of action on this package at the moment, and I'm trying to squeeze in my reviewing before heading to work).
I agree that you should squash it down before merging.
The commit that adds you to the AUTHORS file should be a separate commit. Don't squash that.
In the near future we will remove this separate AUTHORS file anyway, since there is a global gonum AUTHORS file elsewhere.

**Comment by sbinet** [comment]
yes, it is the same.
I'll squash as you advised.
(I am already in the gonum C+A, IIRC)

**Comment by sbinet** [comment]
squashed and polished.

if nobody beats me to (or disagrees) I'll merge this over the week-end.

**Pull request merged**